ci-operator/step-registry/cucushift/chainupgrade/toversion/cucushift-chainupgrade-toversion-commands.sh

```diff
@@ -0,0 +1,773 @@
+release_string="$(< "${SHARED_DIR}/upgrade-edge")"
+# shellcheck disable=SC2207
+TARGET_RELEASES=($(echo "$release_string" | tr ',' ' '))
+echo "Upgrade targets in ci config are ${TARGET_RELEASES[*]}"
+
+export OC="run_command_oc"
+
+mkdir -p /tmp/client
+export OC_DIR="/tmp/client"
+export PATH=${OC_DIR}:$PATH
+for target in "${TARGET_RELEASES[@]}"; do
+    # we need this DUMMY_TARGET_VERSION from ci config to do some pre-check
+    DUMMY_TARGET_VERSION="$(env "NO_PROXY=*" "no_proxy=*" oc adm release info "${target}" --output=json | jq -r '.metadata.version')"
+    TARGET_MAJOR_MINOR_VERSION="$(echo "${DUMMY_TARGET_VERSION}" | cut -d. -f1,2)"
+    TARGET_MINOR_VERSION="$(echo "${DUMMY_TARGET_VERSION}" | cut -d. -f2)"
+    export TARGET_MAJOR_MINOR_VERSION
+    export TARGET_MINOR_VERSION
+    
+    SOURCE_VERSION="$(oc get clusterversion --no-headers | awk '{print $2}')"
+    SOURCE_MINOR_VERSION="$(echo "${SOURCE_VERSION}" | cut -f2 -d.)"
+    export SOURCE_VERSION
+    export SOURCE_MINOR_VERSION
+
+    if ! get_target_from_available_update "${TARGET_MAJOR_MINOR_VERSION}"; then
+        echo "Fail to get target from available update for version: ${TARGET_MAJOR_MINOR_VERSION}"
+        exit 1
+    fi
+
+    echo -e "Source release version is: ${SOURCE_VERSION}; Source minor version is: ${SOURCE_MINOR_VERSION}"
+    echo -e "Target release version is: ${TARGET_VERSION}; Target minor version is: ${TARGET_MINOR_VERSION}"
+    # Target oc will be extract in the /tmp/client directory
+    extract_oc 
+
+    if check_ota_case_enabled "OCP-48683"; then
+        channel=$(oc get clusterversion version -o json|jq -r '.spec.channel')
+        
+        echo "Check updates to target_version in channel-last-hop is the same with channel-current-hop"
+        recommends_last_hop=$(oc get clusterversion version -o json|jq -r '.status.availableUpdates[]?.version'|grep "${TARGET_MAJOR_MINOR_VERSION}")
+        if ! retrieve_updates_by_channel "candidate-${TARGET_MAJOR_MINOR_VERSION}"; then
+            echo "Fail to get latest update through candidate-${TARGET_MAJOR_MINOR_VERSION}"
+            exit 1
+        fi
+        recommends_current_hop=$(oc get clusterversion version -o json|jq -r '.status.availableUpdates[]?.version'|grep "${TARGET_MAJOR_MINOR_VERSION}")
+        if [[ "${recommends_last_hop}" != "${recommends_current_hop}" ]]; then
+            echo "The available update from channel-last-hop:${recommends_last_hop} is different from channel-current-hop: ${recommends_current_hop}."
+            exit 1
+        fi
+
+        ver_in_channel=$(echo "${channel}"|cut -d- -f2)
+        stable_channel="stable-${ver_in_channel}"
+        eus_channel="eus-${ver_in_channel}"
+        # before GA, eus/stable channel will not be promoted, skip the checkpoints.
+        # currently only even number versions supported for eus channel, if odd number versions, skip the checkpoints.
```

Comment:
we do not check odd number versions now, right?

Reply:
Yes, we don't check odd or even directly now. But it still skip odd number versions, if odd versions, it will fail check_channel_enabled.